guides/v2.2/contributor-guide/contributing_dod.md

```diff
@@ -57,74 +57,115 @@ Any backward-incompatible changes must also be recorded in the accompanying docu
+* Broad-form integration tests focus on testing something with less isolation and will involve multiple components explicitly collaborating. (compared to narrow-form where the dependencies are implicit collaborating)
+
+Integration test policy:
+
+* Methods and classes must have blackbox or greybox test coverage making sure to include all variations for normal use as well as corner-cases. 
+* SPI's must have test coverage. Often times Magento has extension points that may only be utilized by extension developers. Use TestModule's to implement those SPI extension points and verify they are used correctly. 
+* As mentioned above, classes marked with `@api` must contain coverage. Integration tests must be used cover these classes unless a unit test is more appropriate.
+* Consumers of default SPI's implementations should have at least basic coverage that ensures the default implementations of the SPI are correctly configured and loaded.  
+  
+  For example, `Magento\Framework\SomeClassFilter` may contain a `FilterPool` that comes with default `FilterInterface`'s from `Magento\Framework`. Each of these implementations would have their own coverage pursuant to this document. However, there should also be some basic assertions within the test coverage for `Magento\Framework\SomeClassFilter` that ensure each of the default filters are loaded correctly.     
+  This shouldn't be explicitly coverage such as `$filter->isLoaded('someDefaultFilter')` 
+
+See: [Running Integration Tests][3].
 
 ### Functional Tests
 
-Must cover new or changed application behavior (functional).
-Added/updated functionality should be covered by a functional autotest(s) related to previous sprint commitment in scope of current sprint.
+UI functional tests are inherently unstable irrespective of platform or testing framework. For this reason along with reasons of maintainability and quality assurance delivery time, these tests should be prioritized below all other tests types.
+
+Functional test policy: 
+* Web API endpoints must have functional test coverage via api-functional tests. These tests should ensure that the endpoints behave in accordance to their service contracts regardless of the actual concrete implementation that may be loaded.
+* For UI-related code, there are some circumstances where it is not possible to cover changes with unit or integration tests. In these cases the appropriate type of coverage would be a functional test but this is a last resort only. 
+  However, aside from those cases UI functional tests should only be used to cover P0/P1 testing scenarios. If there are no scenarios available to make this decision, a product owner must be consulted for approval of new tests.
+  
+  If there is a scenario new or old that isn't classified as a P0/P1, it can be discussed with a product owner to get it elevated if it is important enough to deserve a dedicated test.
```

Comment:
Might not work well with community contributions as it might be problematic to discuss it with product owner
@okorshenko Could you please provide feedback on this ?

Reply:
Hi @xmav, you are right. Community members don't have access to test cases and their status. Also, it is very hard to contact PO for them.  There should be an exemption for community members.

Reply:
@xmav @okorshenko We will be publishing our list of P0/P1 scenarios to the devdocs as the authoritative list for internal and external developers. As for how to contact the PO, we need to create a culture where this is not true. It doesn't make sense to separate the developers that contribute from the owners of the code. If there is a communications gap we should fix it and not compromise our quality.